Gramine build command fails when run without prefix option

<!-- Thanks for reporting a bug.

Please provide as much information as possible.
Paste any logs using three backticks (```).

IMPORTANT: please do not post general questions here;
if you have questions, please use mailing list: gramine-users@googlegroups.com
-->

## Description of the problem
If you follow the steps mentioned in the wiki for building Gramine (which doesn't include any prefix for installation), it would result in a failure at the ninja build step where it's not able to install glibc under `/usr/local`

```
checking for kernel header at least 3.2.0... ok
*** On GNU/Linux systems the GNU C Library should not be installed into
*** /usr/local since this might make your system totally unusable.
*** We strongly advise to use a different prefix.  For details read the FAQ.
*** If you really mean to do this, run configure again using the extra
*** parameter `--disable-sanity-checks'.
ninja: build stopped: subcommand failed.
```

## Steps to reproduce
<!-- NOTE: please specify the exact commit ID on which you reproduced the issue

### PLEASE ENSURE THAT THE ISSUE REPRODUCES ON THE CURRENT MASTER BRANCH ###

-->
```
make
make ISGX_DRIVER_PATH="" SGX=1                  # this assumes Linux 5.11+
meson setup build/ --buildtype=release -Ddirect=enabled -Dsgx=enabled
ninja -C build/
sudo ninja -C build/ install
```

## Expected results
Gramine build and installation should PASS, as the commands are from the documentation.

## Actual results
Ninja build results in a failure.

<!-- ## Additional information -->
<!-- if applicable, uncomment and fill this section -->

--- a/subprojects/packagefiles/glibc-2.34/compile.sh
+++ b/subprojects/packagefiles/glibc-2.34/compile.sh
@@ -42,6 +42,7 @@ mkdir -p "$BUILDDIR"
         --with-tls \
         --without-gd \
         --without-selinux \
+        --disable-sanity-checks \
         --disable-test \
         --disable-nscd
 

